pkg/filesystem/local_directory_linux.go

```diff
@@ -17,6 +22,10 @@ import (
 // rawDeviceNumber is the equivalent of POSIX dev_t.
 type rawDeviceNumber = uint64
 
+var (
```

Comment:
Maybe best to call this as follows?
```
var workingDirectoryLock sync.Mutex
```
Be sure to also add a comment on top, explaining why this exists.